src/Discord.Net.Core/Entities/Interactions/ApplicationCommandOptionChoice.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Discord
+{
+    /// <summary>
+    ///     Represents a choice for a <see cref="IApplicationCommandInteractionDataOption"/>. This class is used when making new commands.
+    /// </summary>
+    public class ApplicationCommandOptionChoiceProperties
+    {
+        private string _name;
+        private object _value;
+        /// <summary>
+        ///     The name of this choice.
+        /// </summary>
+        public string Name
+        {
+            get => _name;
+            set
+            {
+                if(value?.Length > 100)
+                    throw new ArgumentException("Name length must be less than or equal to 100");
+                _name = value;
+            }
+        }
+
+        // Note: discord allows strings & ints as values. how should that be handled?
```

Comment:
How about a class having a constructor for each type?
Would prefer that instead of not knowing about the restriction until I look into my logs and see the exceptions.

Reply:
```csharp
public class StringOrInteger 
{
    public object Value { get; private set; }
    
    public StringOrInteger(int value) => Value = value;
    public StringOrInteger(string value) => Value = value;
    
    public static explicit operator StringOrInteger(int value) => new StringOrInteger(value);
    public static explicit operator StringOrInteger(string value) => new StringOrInteger(value);
}
```